server/src/main/java/org/elasticsearch/search/fetch/FetchPhase.java

```diff
@@ -138,6 +140,28 @@ public void execute(SearchContext context) {
                 fieldsVisitor = new CustomFieldsVisitor(storedToRequestedFields.keySet(), loadSource);
             }
         }
+        int[] docIds = new int[context.docIdsToLoadSize()];
+        int[] sortedDocIds = new int[context.docIdsToLoadSize()];
+        int[] indices = new int[context.docIdsToLoadSize()];
+        Sorter sorter = new IntroSorter() {
+            @Override
+            protected void swap(int i, int j) {
+                int left = docIds[i];
+                docIds[i] = docIds[j];
+                docIds[j] = left;
+            }
+
+            @Override
+            protected void setPivot(int i) {
+
+            }
+
+            @Override
+            protected int comparePivot(int j) {
+                return 0;
+            }
+        };
```

Comment:
why do we need to do this compared to before?

Reply:
That's an unrelated change sorry :(. I removed it from this pr.